### DISCUSSION TITLE
XYChart: Initial Height based on Parent width?

### DISCUSSION THREAD

**Discussion by lapidus** [airbnb/visx#1826] [General]
The XYChart itself is responsive. But would we be a good way of responding to the parent containers width to set the height specifically? 

I'd like to achieve something like this:

```
let height = 500

if(parentWidth < 500){
   height = 300
}

<XYChart
   height={height}
   ...
>
   ...
</XYChart>

```

**Comment by williaster** [comment]
Hey @lapidus, `XYChart` is responsive by default but you can pass your own `width` and `height`. To accomplish what you want I would just wrap it in a `ParentSize` and pass the desired dimensions. Something like this:

```tsx
<ParentSize>
  {({ width, height }) => (
    <XYChart 
      width={width}
      height={/* base on width */}
      {...}
    />
  )}
</ParentSize>
```

**Reply by lapidus** [reply]
Thanks a lot for confirming! I ended up using that solution and it works fine!
